src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -0,0 +1,195 @@
+
+            // Each reference is assigned its own unique id
+            var id = 0;
+            foreach (var (definition, references) in definitionMap)
+            {
+                // Creating the reference item that corresponds to the definition.
+                var definitionItem = await GenerateReferenceItem(
+                    id, definitionId: id, document, position, definition.SourceSpans.FirstOrDefault(), context, definition.DisplayableProperties,
+                    definition.GetClassifiedText(), symbolUsageInfo: null, cancellationToken).ConfigureAwait(false);
+
+                referenceItems.Add(definitionItem);
+                var definitionId = id;
+                id++;
+
+                // Creating a reference item for each reference.
+                foreach (var reference in references)
+                {
+                    var referenceItem = await GenerateReferenceItem(
+                        id, definitionId, document, position, reference.SourceSpan, context, reference.AdditionalProperties,
+                        definitionText: null, reference.SymbolUsageInfo, cancellationToken).ConfigureAwait(false);
+
+                    referenceItems.Add(referenceItem);
+                    id++;
+                };
+            }
+
+            return referenceItems.ToArray();
+
+            async Task<LSP.VSReferenceItem> GenerateReferenceItem(
+                int id,
+                int? definitionId,
+                Document originalDocument,
+                int originalPosition,
+                DocumentSpan documentSpan,
+                SimpleFindUsagesContext context,
+                ImmutableDictionary<string, string> properties,
+                ClassifiedTextElement? definitionText,
+                SymbolUsageInfo? symbolUsageInfo,
+                CancellationToken cancellationToken)
+            {
+                LSP.Location? location = null;
+
+                // If we have no source span, our location may be in metadata.
+                if (documentSpan == default)
+                {
+                    var symbol = await SymbolFinder.FindSymbolAtPositionAsync(originalDocument, originalPosition, cancellationToken).ConfigureAwait(false);
+                    if (symbol != null && symbol.Locations != null && !symbol.Locations.IsEmpty && symbol.Locations.First().IsInMetadata)
+                    {
+                        var declarationFile = await _metadataAsSourceFileService.GetGeneratedFileAsync(originalDocument.Project, symbol, false, cancellationToken).ConfigureAwait(false);
+
+                        var linePosSpan = declarationFile.IdentifierLocation.GetLineSpan().Span;
+                        location = new LSP.Location
+                        {
+                            Uri = new Uri(declarationFile.FilePath),
+                            Range = ProtocolConversions.LinePositionToRange(linePosSpan),
+                        };
+                    }
```

Comment:
what happens if we get here and haven't set hte location?  what is the experience then?

Reply:
Hmm, from testing, it looks like if the location property is empty, no references show up at all. Will push a change to skip definitions and references without locations.